packages/backend-data/src/convert_schema.ts

```diff
@@ -40,24 +82,127 @@ export const convertSchemaToCDK = (
      * to generate that argument for us (so it's consistent with a customer using normal Graphql strings), then
      * apply that value back into the final IAmplifyDataDefinition output for data-schema users.
      */
-    const { schema: transformedSchema, functionSlots } = schema.transform();
+    const {
+      schema: transformedSchema,
+      functionSlots,
+      customSqlDataSourceStrategies,
+    } = schema.transform();
+
+    const provisionStrategyName =
+      stableBackendIdentifiers.getStableBackendHash();
+
+    const dbStrategy = convertDatabaseConfigurationToDataSourceStrategy(
+      schema.data.configuration.database,
+      customSqlDataSourceStrategies,
+      backendSecretResolver,
+      provisionStrategyName
+    );
 
     const generatedModelDataSourceStrategies = AmplifyDataDefinition.fromString(
       transformedSchema,
-      {
-        dbType,
-        provisionStrategy,
-      }
+      dbStrategy
     ).dataSourceStrategies;
+
+    if (dbStrategy.dbType === 'DYNAMODB') {
+      return {
+        schema: transformedSchema,
+        functionSlots,
+        dataSourceStrategies: generatedModelDataSourceStrategies,
+      };
+    }
+
     return {
       schema: transformedSchema,
       functionSlots,
       dataSourceStrategies: generatedModelDataSourceStrategies,
+      customSqlDataSourceStrategies:
+        customSqlDataSourceStrategies?.map(
+          (existing: CustomSqlDataSourceStrategy) => ({
+            ...existing,
+            strategy: dbStrategy,
+          })
+        ) || [],
+    };
+  }
+
+  return AmplifyDataDefinition.fromString(schema, DYNAMO_DATA_SOURCE_STRATEGY);
+};
+
+/**
+ * Given an input list of CDK Graphql Def interface schemas, produce the relevant CDK Graphql Def interface
+ * @param schemas the cdk graphql definition interfaces to combine
+ * @returns the cdk graphql definition interface
+ */
+export const combineCDKSchemas = (
+  schemas: IAmplifyDataDefinition[]
+): IAmplifyDataDefinition => {
+  return AmplifyDataDefinition.combine(schemas);
+};
+
+/**
+ * Given the generated rds builder database configuration, convert it into the DataSource strategy
+ * @param configuration the database configuration from `data-schema`
+ * @returns the data strategy needed to configure the data source
+ */
+const convertDatabaseConfigurationToDataSourceStrategy = (
+  configuration: DataSourceConfiguration,
+  customSqlDataSourceStrategies: CustomSqlDataSourceStrategy[] = [],
+  backendSecretResolver: BackendSecretResolver,
+  provisionStrategyName: string
+): ModelDataSourceStrategy => {
+  if (configuration.engine === 'dynamodb') {
+    return DYNAMO_DATA_SOURCE_STRATEGY;
+  }
```

Comment:
CDK construct supports the old-style CloudFormation provisioning strategy, as well as the new style Amplify-managed tables. `defineData` defaults to Amplify-managed tables, but if customers can override that, this is not a safe check. Even if you don't provide a builder for it, the fact that `combineCDKSchemas` accepts an `IAmplifyDataDefinition` means that it's at least technically possible for a caller to supply [the `DefaultDynamoDbModelDataSourceStrategy`](https://github.com/aws-amplify/amplify-category-api/blob/fd7f6fbc17c199331c4b04debaff69ea0424cd74/packages/amplify-graphql-api-construct/src/model-datasource-strategy-types.ts#L30-L33) rather than [the `AmplifyDynamoDbModelDataSourceStrategy`](https://github.com/aws-amplify/amplify-category-api/blob/fd7f6fbc17c199331c4b04debaff69ea0424cd74/packages/amplify-graphql-api-construct/src/model-datasource-strategy-types.ts#L38-L41)

Reply:
`combineCDKSchemas` is not exported outside of this module / not exposed to customers. It is only invoked from `backend-data/src/factory.ts` and the `IAmplifyDataDefinition` is only crafted internally as well.